flake.nix

```diff
@@ -20,59 +24,109 @@
     };
   };
 
-  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, nix-npm-buildpackage, flake-utils }:
-    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
-      let
-        inherit (nixpkgs.lib) foldl' composeExtensions;
-
-        pkgs = nixpkgs.legacyPackages.${system}.extend
-          (foldl' composeExtensions (_: _: { }) [
-            nix-npm-buildpackage.overlay
-            haskell-nix.overlay
-          ]);
+  outputs = { self, nixpkgs, haskell-nix, serokell-nix, deploy-rs, hackage, stackage, nix-npm-buildpackage, flake-utils }:
+  nixpkgs.lib.recursiveUpdate
+  (let
+    system = "x86_64-linux";
+    inherit (nixpkgs.lib) mapAttrs concatMapStrings escapeShellArg;
+    inherit (nixpkgs.legacyPackages.${system}) runCommandNoCC;
+  in {
+    # Do not roll back profile closure deployment
+    deploy.magicRollback = false;
 
-        backend = pkgs.callPackage ./backend { };
-        frontend = pkgs.callPackage ./frontend { };
-        frontendCheck = checkPhase:
-          frontend.package.overrideAttrs (_: {
-            doBuild = false;
-            doCheck = true;
-            inherit checkPhase;
-          });
+    deploy.nodes.castor =
+      let
+        # pkgs.symlinkJoin et al can only handle folders, not files; so we roll
+        # our own, but skip the `dirname` call.
```

Comment:
There is `linkFarm`, which can handle files

Reply:
I tried glinkFarm` and it complained about it being a file.

Reply:
Nevermind, seems I'm just dumb.